From ec5fa2931e4f16e06e17f4facbba6e13bc260e14 Mon Sep 17 00:00:00 2001
From: jozwright <jozwright@gmail.com>
Date: Mon, 1 Dec 2014 15:37:28 -0800
Subject: [PATCH 05/23] data saves in psql

---
 main.rb | 8 ++++++--
 1 file changed, 6 insertions(+), 2 deletions(-)

diff --git a/main.rb b/main.rb
index 952c47f..951e08f 100644
--- a/main.rb
+++ b/main.rb
@@ -1,9 +1,9 @@
 require 'pg'
 require 'sinatra'
 require 'sinatra/activerecord'
-require 'json'
 
 set :database, 'postgres://localhost/deal_a_day'
+DB = PG.connect({:dbname => 'deal_a_day'})
 
 class Detail < ActiveRecord::Base
 end
@@ -16,5 +16,9 @@ post '/' do
   File.open('uploads/' + params['upload'][:filename], "w") do |f|
     f.write(params['upload'][:tempfile].read)
   end
-  return "Your data was successfully uploaded!"
+
+  file_name = File.absolute_path('uploads/' + params['upload'][:filename]) 
+  DB.exec("COPY details(purchaser_name, description, price, amount, address, merchant_name) FROM '#{file_name}' WITH CSV HEADER DELIMITER AS ',';")
+  
+  return "Your file was successfully uploaded!"
 end
\ No newline at end of file
-- 
1.8.4

