From 68f239b88a65c8123cbb57ca5d6135031ec7bd91 Mon Sep 17 00:00:00 2001
From: jozwright <jozwright@gmail.com>
Date: Tue, 2 Dec 2014 13:07:18 -0800
Subject: [PATCH 08/23] calculate revenue from last upload

---
 db/schema.rb |  2 +-
 main.rb      | 26 +++++++++++++++++---------
 2 files changed, 18 insertions(+), 10 deletions(-)

diff --git a/db/schema.rb b/db/schema.rb
index eacb235..c3ad931 100644
--- a/db/schema.rb
+++ b/db/schema.rb
@@ -19,7 +19,7 @@ ActiveRecord::Schema.define(version: 20141201193952) do
   create_table "details", force: true do |t|
     t.string   "purchaser_name"
     t.string   "description"
-    t.decimal  "price",          precision: 8, scale: 2
+    t.decimal  "price", precision: 8, scale: 2
     t.integer  "amount"
     t.string   "address"
     t.string   "merchant_name"
diff --git a/main.rb b/main.rb
index b7c9d2c..ebcf382 100644
--- a/main.rb
+++ b/main.rb
@@ -8,10 +8,22 @@ DB = PG.connect({:dbname => 'deal_a_day'})
 enable :sessions
 
 class Detail < ActiveRecord::Base
-  def self.total_revenue
+  def self.upload(file)
+    File.open('uploads/' + file[:filename], "w") do |f|
+      f.write(file[:tempfile].read)
+    end
+    file_name = File.absolute_path('uploads/' + file[:filename]) 
+    DB.exec("COPY details(purchaser_name, description, price, amount, address, merchant_name) FROM '#{file_name}' WITH CSV HEADER DELIMITER AS ',';")
+    DB.exec("UPDATE details SET created_at = NOW() WHERE created_at IS NULL;")
+  end
+  
+  def self.latest_upload_revenue
     total = 0
+    latest_upload = Detail.order('created_at').last
     Detail.all.each do |d|
-      total += d.price * d.amount
+      if d.created_at == latest_upload.created_at
+        total += d.price * d.amount
+      end
     end
     total
   end
@@ -22,13 +34,9 @@ get '/' do
 end
 
 post '/' do
-  if params[:upload]
-    File.open('uploads/' + params['upload'][:filename], "w") do |f|
-      f.write(params['upload'][:tempfile].read)
-    end
-    file_name = File.absolute_path('uploads/' + params['upload'][:filename]) 
-    upload = DB.exec("COPY details(purchaser_name, description, price, amount, address, merchant_name) FROM '#{file_name}' WITH CSV HEADER DELIMITER AS ',';")
-    flash[:success] = "The total amount of revenue is $#{Detail.total_revenue}0. Lookin' good!"
+  if params['upload']
+    Detail.upload(params['upload'])
+    flash[:success] = "The total amount of revenue is $#{Detail.latest_upload_revenue}0. Lookin' good!"
   else
     flash[:error] = "You did not select a file to upload."
   end
-- 
1.8.4

